arithmetization/src/main/java/net/consensys/linea/zktracer/ZkCounter.java

```diff
@@ -388,12 +397,18 @@ public ZkCounter(LineaL1L2BridgeSharedConfiguration bridgeConfiguration, Fork fo
     moduleToCount = Stream.concat(checkedModules().stream(), uncheckedModules().stream()).toList();
 
     log.info("[ZkCounter] Created ZkCounter for fork {}", fork);
+    if (!countHistoricalBlockHashes) {
+      log.info(
+          "[ZkCounter] Historical BlockHashes not counted for BLOCKHASH, might miss 256*6 lines that are common for the whole conflation.");
+    }
   }
 
   @Override
   public void traceStartConflation(long numBlocksInConflation) {
-    blockHash.updateTally(
-        (int) ((BLOCKHASH_MAX_HISTORY + numBlocksInConflation) * NB_ROWS_BLOCKHASH));
+    if (countHistoricalBlockHashes) {
+      blockHash.updateTally(
+          (int) ((BLOCKHASH_MAX_HISTORY + numBlocksInConflation) * NB_ROWS_BLOCKHASH));
+    }
```

Comment:
Tiny comment / sanity check: the final block of the conflation shouldn't have a corresponding operation in the `BLOCK_HASH` module, correct ? So shouldn't the formula be
```java
(BLOCKHASH_MAX_HISTORY + numBlocksInConflation - 1) * NB_ROWS_BLOCKHASH
```
?